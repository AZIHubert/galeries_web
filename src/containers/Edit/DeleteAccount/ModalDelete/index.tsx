import { useFormik } from 'formik';
import * as React from 'react';
import {
  useSelector,
} from 'react-redux';

import Button from '#components/Button';
import Field from '#components/Field';
import Text from '#components/Text';

import { deleteAccountSchema } from '#helpers/schemas';

import { loadingSelector } from '#store/selectors';

const initialValues = {
  deleteAccountSentence: '',
  password: '',
  userNameOrEmail: '',
};

const ModalDelete = () => {
  const formik = useFormik({
    initialValues,
    onSubmit: () => {},
    validateOnBlur: true,
    validateOnChange: false,
    validationSchema: deleteAccountSchema,
  });
  const loading = useSelector(loadingSelector);
  return (
    <div>
      <Text
        color='danger'
        styles={{
          fontSize: 1.1,
          marginBottom: 30,
        }}
      >
        Are you sure you want to do this?
      </Text>
      <form onSubmit={formik.handleSubmit}>
        <Field
          disabled={loading}
          error={formik.errors.userNameOrEmail}
          fieldTestId='field'
          id='userNameOrEmail'
          label='user name or email'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          styles={{
            marginBottom: 6,
          }}
          stylesLaptopL={{
            marginBottom: 10,
          }}
          touched={formik.touched.userNameOrEmail}
          type='text'
          value={formik.values.userNameOrEmail}
        />
        <Field
          disabled={loading}
          error={formik.errors.deleteAccountSentence}
          fieldTestId='field'
          id='deleteAccountSentence'
          label='To verify, type "delete my account" below:'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          styles={{
            marginBottom: 6,
          }}
          stylesLaptopL={{
            marginBottom: 10,
          }}
          touched={formik.touched.deleteAccountSentence}
          type='text'
          value={formik.values.deleteAccountSentence}
        />
        <Field
          disabled={loading}
          error={formik.errors.password}
          fieldTestId='field'
          id='password'
          label='confirm your password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          touched={formik.touched.password}
          type='password'
          value={formik.values.password}
        />
        <Button.Default
          danger
          disabled={loading}
          data-testid='submitButton'
          styles={{
            marginBottom: 10,
            marginTop: 20,
          }}
          stylesLaptopL={{
            marginTop: 24,
          }}
          type='submit'
          title='delete your account'
        />
      </form>
    </div>
  );
};

export default ModalDelete;
