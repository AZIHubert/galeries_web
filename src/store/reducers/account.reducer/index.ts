import { ACCOUNT_SET } from '#store/actions';

interface InitialStateI {
  errors: form.AccountI,
  status: store.Status,
}

const initialState: InitialStateI = {
  errors: {
    deleteAccountSentence: '',
    password: '',
    userNameOrEmail: '',
  },
  status: 'pending',
};

export default (
  state = initialState,
  action: store.ActionI,
) => {
  switch (action.type) {
    case ACCOUNT_SET:
      return {
        ...state,
        ...action.payload ? action.payload.data : undefined,
        errors: {
          ...state.errors,
          ...action.payload ? action.payload.data.errors : undefined,
        },
      };
    default:
      return state;
  }
};
