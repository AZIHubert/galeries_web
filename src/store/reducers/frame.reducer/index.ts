import { FRAME_SET } from '#store/actions';

interface InitialStateI {
  status: store.Status;
}

const initialState: InitialStateI = {
  status: 'pending',
};

export default (
  state = initialState,
  action: store.ActionI,
) => {
  switch (action.type) {
    case FRAME_SET:
      return {
        ...state,
        ...action.payload ? action.payload.data : undefined,
      };
    default:
      return state;
  }
};
