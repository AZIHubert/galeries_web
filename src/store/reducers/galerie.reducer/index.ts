import { GALERIE_SET } from '#store/actions';

interface InitialStateI {
  status: store.Status;
  errors: {
    name: string;
  }
}

const initialState: InitialStateI = {
  status: 'pending',
  errors: {
    name: '',
  },
};

export default (
  state = initialState,
  action: store.ActionI,
) => {
  switch (action.type) {
    case GALERIE_SET:
      return {
        ...state,
        ...action.payload ? action.payload.data : undefined,
        errors: {
          ...state.errors,
          ...action.payload ? action.payload.data.errors : undefined,
        },
      };
    default:
      return state;
  }
};
