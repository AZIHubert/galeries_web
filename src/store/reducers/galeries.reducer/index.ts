import { GALERIES_SET } from '#store/actions';

interface InitialStateI {
  end: boolean;
  page: number;
  galeries: { [name: string]: GalerieI };
  status: store.Status;
}

const initialState: InitialStateI = {
  end: false,
  page: 0,
  galeries: {},
  status: 'pending',
};

export default (
  state = initialState,
  action: store.ActionI,
) => {
  switch (action.type) {
    case GALERIES_SET:
      return {
        ...state,
        ...action.payload ? action.payload.data : undefined,
      };
    default:
      return state;
  }
};
