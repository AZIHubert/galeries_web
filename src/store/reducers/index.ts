import { combineReducers } from 'redux';

import login from './login.reducer'; // loginError
import notification from './notification.reducer';
import sendConfirmation from './sendConfirmation.reducer'; // sendConfirmationError
import sendResetPassword from './sendResetPassword.reducer'; // sendResetPasswordError
import signin from './signin.reducer'; // SigninError
import user from './user.reducer';
import ui from './ui.reducer';

export default combineReducers({
  login,
  notification,
  sendConfirmation,
  sendResetPassword,
  signin,
  ui,
  user,
});
