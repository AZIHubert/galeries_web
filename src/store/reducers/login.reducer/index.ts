import { LOGIN_SET } from '#store/actions';

interface InitialStateI {
  errors: form.LoginI,
  status: store.FormStatus,
}

const initialState: InitialStateI = {
  errors: {
    password: '',
    userNameOrEmail: '',
  },
  status: 'pending',
};

export default (
  state = initialState,
  action: store.ActionI,
) => {
  switch (action.type) {
    case LOGIN_SET:
      return {
        ...state,
        ...action.payload ? action.payload.data : undefined,
        errors: {
          ...state.errors,
          ...action.payload ? action.payload.data.errors : undefined,
        },
      };
    default:
      return state;
  }
};
