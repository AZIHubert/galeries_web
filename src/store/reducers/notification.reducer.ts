import { SET_NOTIFICATION } from '../actions';

const initialState = {
  text: '',
  error: false,
};

export default (
  notification = initialState,
  action: ActionNotificationI,
) => {
  const { payload } = action;
  switch (action.type) {
    case SET_NOTIFICATION:
      return {
        ...notification,
        text: payload.text,
        error: payload.error,
      };
    default:
      return notification;
  }
};
