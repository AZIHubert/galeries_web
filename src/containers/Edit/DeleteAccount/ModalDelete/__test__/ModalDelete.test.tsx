import {
  cleanup,
  fireEvent,
  render,
  screen,
} from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import ThemeProvider from '#contexts/ThemeContext';
import {
  REQUIRED,
} from '#helpers/formErrors';
import reducers from '#store/reducers';

import ModalDelete from '../index';

const mockedStore = createStore(reducers);

const Container = () => (
  <Provider store={mockedStore}>
    <ThemeProvider>
      <ModalDelete />
    </ThemeProvider>
  </Provider>
);

describe('ModalDelete', () => {
  afterEach(cleanup);
  describe('should display error if', () => {
    describe('if password', () => {
      it('is empty', async () => {
        const { getByTestId } = render(<Container />);
        fireEvent.blur(getByTestId('field'));
        const error = await screen.findByText(REQUIRED);
        expect(error).toBeTruthy();
      });
    });
  });
});
