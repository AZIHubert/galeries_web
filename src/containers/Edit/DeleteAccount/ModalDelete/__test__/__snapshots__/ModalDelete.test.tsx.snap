// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDelete renders without crashing 1`] = `
<div
  data-testid="deleteModal"
>
  <p>
    Are you sure you want to do this?
  </p>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="password"
    >
      confirm yout password
    </label>
    <input
      data-testid="passwordField"
      disabled={false}
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
      value=""
    />
    <button
      data-testid="submitButton"
      disabled={false}
      type="submit"
    >
      delete your account
    </button>
  </form>
</div>
`;
